serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc

```diff
@@ -1 +1,269 @@
+
+.Example Order fulfillment Saga Workflow
+image::use-cases/orchestration-based-saga-pattern/order-fulfillment-saga-workflow.png[]
+
+In the workflow diagram you can see how to describe the calls to the participants, the possible errors that each service can throw, and finally, how to define compensations to each step to be executed once an error happens during workflow execution.
+
+=== Saga participants interactions
+
+To define the interactions among participants in the Saga using Serverless Workflow Specification, you can use link:{spec_doc_url}#workflow-states[states] with transitions link:{spec_doc_url}#Transitions[transitions] corresponding to the participants' interactions.
+
+Each state represents a step to be achieved in the Saga and an action associated with the state represents how to invoke a participant to execute a given step, for instance, `processPayment` state has the `processPaymentAction` action responsible to invoke the function to execute the payment processing, that might be done by the participant `Payment Service`.
+The `transition` attribute represents the next step to be started, to perform the order shipping scheduling for the Saga.
+
+[#payment-state]
+.Example the state declaration representing the process payment step in the Saga
+[source,json]
+----
+{
+   "name":"processPayment",
+   "type":"operation",
+   "actions":[
+      {
+         "name":"processPaymentAction",
+         "functionRef":{
+            "refName":"processPayment",
+            "arguments":{
+               "orderId":".orderId",
+               "failService":".failService"
+            }
+         },
+         "actionDataFilter":{
+            "results":".",
+            "toStateData":".paymentResponse"
+         }
+      }
+   ],
+   "transition":"scheduleShipping",
+   "compensatedBy":"CancelPayment",
+   "onErrors":[
+      {
+         "errorRef":"process payment failed",
+         "transition":"ServiceError"
+      }
+   ]
+}
+----
+
+=== Compensations
+
+The core functionality when designing the Saga is the compensation for each step to be executed by a participant.
+In serverless workflow, each state should define how it should be compensated using its compensatedBy property, which indicates another state responsible to perform the compensation for it, i.e. processPayment defines the CancelPayment as the compensation, as define in the <<payment-state, payment state>>.
+
+[source,json]
+"compensatedBy": "CancelPayment"
+
+=== Errors
+
+Compensations are triggered as soon as an error occurs, according to the errors definitions associated states on serverless workflow, where they are identified by a name, for instance, on `processPayment` there is the error `process payment failed` which is a reference to the <<#errors-definitions, errors section>> in the workflow definition.
```

Comment:
Can we break this paragraph a little bit?

Reply:
+1